### PR TITLE
Issue & revoke credentials added

### PR DESCRIPTION
**Pull request by NachoPal**
Solves: #64

### CODE REVIEW

File: pallets/did/src/lib.rs [hunk 1]
```diff
@@ -410,22 +416,52 @@ pub mod pallet {
 		#[pallet::weight(1000000)]
 		pub fn issue_credentials(
```
Comment (peterwht) [inline pallets/did/src/lib.rs:417]: `issue_credentials` never writes anything to storage? Is this expected?

File: pallets/did/src/lib.rs [hunk 2]
```diff
@@ -410,22 +416,52 @@ pub mod pallet {
...
+				did: subject_did,
+				credentials,
+				verifiable_credential_hash,
+			});
+			Ok(())
+		}
+
+		#[pallet::weight(1000000)]
+		pub fn revoke_credentials(
+			origin: OriginFor<T>,
```
Comment (peterwht) [inline pallets/did/src/lib.rs:448]: `revoke_credentials` does not write to storage either.

File: pallets/did/src/lib.rs [hunk 3]
```diff
@@ -410,22 +416,52 @@ pub mod pallet {
...
+			subject_did: DidIdentifierOf<T>,
+			credentials: Vec<CredentialOf<T>>,
+			verifiable_credential_hash: HashOf<T>,
+		) -> DispatchResult {
+			let controller = ensure_signed(origin)?;
+
+			// Ensure origin is the issuer's controller
+			let document = Did::<T>::get(&issuer_did).ok_or(Error::<T>::DidNotFound)?;
+			Self::ensure_controller(controller, &document)?;
+
```
Comment (peterwht) [inline pallets/did/src/lib.rs:459]: Check if `subject_did` exists too.

**Comment by peterwht** [comment]
Does the credential type / validity need to be checked as well?

File: pallets/did/src/lib.rs (hunk 1 above)
Comment (NachoPal) [reply pallets/did/src/lib.rs:417]: Yes, event is listened and stored off-chain

File: pallets/did/src/lib.rs (hunk 2 above)
Comment (NachoPal) [reply pallets/did/src/lib.rs:448]: Yes, event is listened and stored off-chain

File: pallets/did/src/lib.rs (hunk 3 above)
Comment (NachoPal) [reply pallets/did/src/lib.rs:459]: I am not checking `subject_id` because the `did` might not exist anymore

**Comment by NachoPal** [comment]
> Does the credential type / validity need to be checked as well?

Same reason, it might not exist at the time of revoking

**Pull request merged**